pkg/yang/types_builtin.go

```diff
@@ -790,16 +804,41 @@ func (r YRange) Valid() bool {
 // A YangRange is a set of non-overlapping ranges.
 type YangRange []YRange
 
-// ParseRanges parses s into a series of ranges.  Each individual range is
-// in s is separated by the pipe character (|).  The min and max value of
-// a range are separated by "..".  An error is returned if the range is
-// invalid.  The resulting range is sorted and coalesced.
-func ParseRanges(s string) (YangRange, error) {
+// ParseRangesInt parses s into a series of ranges. Each individual range is in s
+// is separated by the pipe character (|).  The min and max value of a range
+// are separated by "..".  An error is returned if the range is invalid. The
+// output range is sorted and coalesced.
+func ParseRangesInt(s string) (YangRange, error) {
+	return parseRanges(s, false, 0)
+}
+
+// ParseRangesDecimal parses s into a series of ranges. Each individual range is in s
+// is separated by the pipe character (|).  The min and max value of a range
+// are separated by "..".  An error is returned if the range is invalid. The
+// output range is sorted and coalesced.
+func ParseRangesDecimal(s string, fracDigRequired uint8) (YangRange, error) {
```

Comment:
I guess I'm not quite clear why we need `fracDigRequired` here. Could you explain?

Here's why I ask. If we have:

```yang
leaf foo {
  type decimal64 {
    fraction-digits 10;
    range "1...3.14159";
 }
```

then this should be valid input. In the ranges we only really care about the upper and lower bounds, which don't need to have fixed precisions. The usage example in RFC6020 shows this kind of input as an example ([here](https://tools.ietf.org/html/rfc6020#section-9.3.5)), so I figure we should support it. If we assume that we don't need `fraction-digits` specified here, then we can keep the public API of `ParseRanges`.

Reply:
The problem is if you have
```
leaf foo {
  type decimal64 {
    fraction-digits 1;
    range "1...3.14159";
 }
```
Then it should fail, but the current code doesn't fail because it automatically deduces the fraction-digits and ignores the value in the YANG file.